core/account-id/src/lib.rs

```diff
@@ -64,6 +64,24 @@ pub use errors::{ParseAccountError, ParseErrorKind};
 #[derive(Eq, Ord, Hash, Clone, Debug, PartialEq, PartialOrd)]
 pub struct AccountId(Box<str>);
 
+#[derive(PartialEq)]
+pub enum AccountType {
+    NamedAccount,
+    NearImplicitAccount,
+    EthImplicitAccount,
```

Comment:
It's not really a refactor if you're adding new type of account :) 
Can you either remove the EthImplicitAccount and keep this PR a pure refactoring or adjust the title and summary?

Reply:
How about now? The intention for this PR is to aggregate changes that do not impact any result of code execution.